### PR TITLE
Feature/customers

### PR DESCRIPTION
**Pull request by klvnch**
**Create Customer Modul**
The change:
- Created Customer Model, Factory and add the factory to the DatabaseSeeder.
- Created Customer folder for 3 Vue pages (Create, Edit, and Index).
- Created Controller that is responsible for all customer pages and processes.
- Add model relationship to Account, and Contact because Account and Contact are relate to Customer
- Changed Contact Controller to show the customer data based on contact's id.
- Add a table of Customer on Edit Contact page.
- Add new routes for Customer pages.
- Add 1 more menu on MainMenu for Customer menu.

### CODE REVIEW

File: app/Models/Account.php [hunk 1]
```diff
@@ -20,4 +20,8 @@ public function contacts()
     {
         return $this->hasMany(Contact::class);
     }
+
+    public function customer(){
+        return $this->hasMany(Customer::class);
+    }
```
Comment (umaraziz0) [inline app/Models/Account.php]: yang kaya ini coba di reformat ya, biar spacing kurung2nya sama kaya yang lain.
harusnya tinggal di save lagi aja kalo autoformat udah nyala

File: app/Models/Contact.php [hunk 2]
```diff
@@ -21,6 +21,10 @@ public function organization()
         return $this->belongsTo(Organization::class);
     }
 
+    public function customer(){
+        return $this->belongsTo(Customer::class);
+    }
```
Comment (umaraziz0) [inline app/Models/Contact.php]: reformat

File: app/Models/Customer.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public function resolveRouteBinding($value, $field = null)
+    {
+        return $this->where($field ?? 'id', $value)->withTrashed()->firstOrFail();
+    }
+
+    protected $dates = ['deleted_at'];
+
+    public function contacts(){
+        return $this->hasMany(Contact::class);
+    }
```
Comment (umaraziz0) [inline app/Models/Customer.php]: reformat

**Comment by klvnch** [comment]
Code sudah saya format ulang sesuai dengan autoformat

**Review by umaraziz0 — APPROVED** [review]
oke sip thanks

**Pull request merged**